### PR TITLE
tests/thread_flood: improve test output

### PR DESCRIPTION
**Pull request by aabadie**
While performing 'manual' testing for the release, I realized that some testing script were missing or were outdated.
This PR is a try to cleanup one of them. Comments welcome.

### CODE REVIEW

File: tests/thread_flood/tests/test_thread.py [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
+import sys
 
-term = pexpect.spawn("make term")
+sys.path.append(os.path.join(os.environ['RIOTBASE'], 'dist/tools/testrunner'))
+import testrunner
 
-term.expect('Start spawning\r\n')
-term.expect('Thread creation successful aborted\r\n')
+def testfunc(child):
+    child.expect(u'[SUCCESS].*')
```
Comment (cladmi) [inline tests/thread_flood/tests/test_thread.py]: I would keep the full output here in the expect calls.
This way, the python test is self explanatory even if it duplicates the strings.

**Comment by cladmi** [comment]
I would also like to see a separate commit saying "migrate to testrunner".

**Comment by aabadie** [comment]
@cladmi, I addressed your comments

**Comment by aabadie** [comment]
Side question: is there a battle plan for migrating actual tests to use the testrunner ? I think I can help if needed.

**Comment by cladmi** [comment]
As some measures are needed when testing on boards, I think it is a good idea to use the same tool for all tests.

**Comment by aabadie** [comment]
> As some measures are needed when testing on boards, I think it is a good idea to use the same tool for all tests.

I'm not sure to understand what you mean by this.

**Comment by aabadie** [comment]
> Side question: is there a battle plan for migrating actual tests to use the testrunner ? I think I can help if needed.

So no answer to this ?

**Comment by cladmi** [comment]
I just meant, its better to use 'testrunner' so there is only one tool where all side problems can be solved.

And currently no battle plans I think, there is the release first.

File: tests/thread_flood/tests/test_thread.py [hunk 2]
```diff
@@ -1,8 +1,13 @@
 #!/usr/bin/python
```
Comment (haukepetersen) [inline tests/thread_flood/tests/test_thread.py]: minor thing: for the sake of continuity, how about we rename this file to `01-run.py` to be consistent?

Comment (haukepetersen) [reply tests/thread_flood/tests/test_thread.py]: Also: change to `#!/usr/bin/env python3` (as my 'old' mint has trouble with its old 2.7.x...), thx!

File: tests/thread_flood/tests/test_thread.py [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
+import sys
 
-term = pexpect.spawn("make term")
+sys.path.append(os.path.join(os.environ['RIOTBASE'], 'dist/tools/testrunner'))
+import testrunner
 
-term.expect('Start spawning\r\n')
-term.expect('Thread creation successful aborted\r\n')
+def testfunc(child):
+    child.expect(u'[SUCCESS] Thread creation successfully aborted')
```
Comment (haukepetersen) [inline tests/thread_flood/tests/test_thread.py]: this does not work: either you have to escape the `[ ]` brackets, or better use 
`child.expect_exact("[SUCCESS] Thread creation successfully aborted")`
here.

**Comment by haukepetersen** [comment]
Forgot to mention: would be nice if we could merge this ASAP to get it into RC2!

File: tests/thread_flood/tests/test_thread.py (hunk 3 above)
Comment (aabadie) [reply tests/thread_flood/tests/test_thread.py]: Thanks ! Will change that.

**Comment by aabadie** [comment]
> And currently no battle plans I think, there is the release first.

Even if it add extra organisation stuff, I think it would be nice to have an issue for tracking all that, since it requires quite a few things to change (in a very short time and during a RC step). It will simplify the synchronization between participants and avoid overlap.

**Comment by haukepetersen** [comment]
@aabadie can you fix this test now(ish)?

**Comment by haukepetersen** [comment]
ups, did not see your comment above that you are already on it, so never mind :-)

**Comment by aabadie** [comment]
Done

**Comment by aabadie** [comment]
And tested locally on nucleo-l073, now it should works

**Comment by haukepetersen** [comment]
and go.

**Comment by haukepetersen** [comment]
@aabadie could you please provide a backport into the `2017.10-branch`?! Thx.

**Comment by miri64** [comment]
backport provided in #7839

**Pull request merged**